### PR TITLE
1.10 d-man delivery fix

### PR DESCRIPTION
**Pull request by xavierh**
* Fix forceTransferStack return value

* Fix javadoc

* Fix happiness calculation about housing

* Transfer for real instead of simulating

* remove some warnings

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java [hunk 1]
```diff
@@ -220,7 +220,7 @@ public Object handle(@NotNull final World world, @NotNull final BlockPos pos, @N
     public static class FlowerPotPlacementHandler implements IPlacementHandler
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java:220]: MAJOR Reduce this class from 31 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)

File: src/main/java/com/minecolonies/coremod/items/ItemScepterPermission.java [hunk 2]
```diff
@@ -133,7 +133,7 @@ public EnumActionResult onItemUse(
 
         toggleItemMode(playerIn, compound);
 
-        return new ActionResult(EnumActionResult.SUCCESS, scepter);
+        return new ActionResult<ItemStack>(EnumActionResult.SUCCESS, scepter);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/items/ItemScepterPermission.java]: MINOR Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2293)

File: src/main/java/com/minecolonies/coremod/items/ItemScepterPermission.java [hunk 3]
```diff
@@ -122,7 +122,7 @@ public EnumActionResult onItemUse(
     {
         if (worldIn.isRemote)
         {
-            return new ActionResult(EnumActionResult.SUCCESS, scepter);
+            return new ActionResult<ItemStack>(EnumActionResult.SUCCESS, scepter);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/items/ItemScepterPermission.java]: MINOR Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2293)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 100 issues
* ![MAJOR][MAJOR] 44 major
* ![MINOR][MINOR] 48 minor
* ![INFO][INFO] 8 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/eac30293cab95fee287c6c98e2b6e2d982bd8c5b/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.233 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/eac30293cab95fee287c6c98e2b6e2d982bd8c5b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 43 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/eac30293cab95fee287c6c98e2b6e2d982bd8c5b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): class "Colony" has 82 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/eac30293cab95fee287c6c98e2b6e2d982bd8c5b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): Refactor this class so it has no more than 20 fields, rather than the 36 it currently has. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1820)
1. ![MAJOR][MAJOR] [Colony.java#L641](https://github.com/Minecolonies/minecolonies/blob/eac30293cab95fee287c6c98e2b6e2d982bd8c5b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L641): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1159](https://github.com/Minecolonies/minecolonies/blob/eac30293cab95fee287c6c98e2b6e2d982bd8c5b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1159): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1160](https://github.com/Minecolonies/minecolonies/blob/eac30293cab95fee287c6c98e2b6e2d982bd8c5b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1160): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1273](https://github.com/Minecolonies/minecolonies/blob/eac30293cab95fee287c6c98e2b6e2d982bd8c5b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1273): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1275](https://github.com/Minecolonies/minecolonies/blob/eac30293cab95fee287c6c98e2b6e2d982bd8c5b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1275): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1277](https://github.com/Minecolonies/minecolonies/blob/eac30293cab95fee287c6c98e2b6e2d982bd8c5b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1277): Assign this magic number 128 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 98 issues
* ![MAJOR][MAJOR] 44 major
* ![MINOR][MINOR] 46 minor
* ![INFO][INFO] 8 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/3cfb2395d387f608ab946e956afcf2709735105e/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.233 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/3cfb2395d387f608ab946e956afcf2709735105e/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 43 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/3cfb2395d387f608ab946e956afcf2709735105e/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): class "Colony" has 82 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/3cfb2395d387f608ab946e956afcf2709735105e/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): Refactor this class so it has no more than 20 fields, rather than the 36 it currently has. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1820)
1. ![MAJOR][MAJOR] [Colony.java#L641](https://github.com/Minecolonies/minecolonies/blob/3cfb2395d387f608ab946e956afcf2709735105e/src/main/java/com/minecolonies/coremod/colony/Colony.java#L641): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1159](https://github.com/Minecolonies/minecolonies/blob/3cfb2395d387f608ab946e956afcf2709735105e/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1159): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1160](https://github.com/Minecolonies/minecolonies/blob/3cfb2395d387f608ab946e956afcf2709735105e/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1160): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1273](https://github.com/Minecolonies/minecolonies/blob/3cfb2395d387f608ab946e956afcf2709735105e/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1273): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1275](https://github.com/Minecolonies/minecolonies/blob/3cfb2395d387f608ab946e956afcf2709735105e/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1275): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1277](https://github.com/Minecolonies/minecolonies/blob/3cfb2395d387f608ab946e956afcf2709735105e/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1277): Assign this magic number 128 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by marvin-bitterlich** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/986/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by Raycoms** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/986/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Pull request merged**